### ISSUE TITLE
About val_ data and test_data

### ISSUE THREAD

**Issue by J-byte-cloud** [openvinotoolkit/anomalib#539]
In the anomalib framework,  what's the difference between val_ data and test_data? I has read the code, val_ data was randomly extracted from test_data ,and the were not related to train_ data, so what were their respective roles?

**Comment by djdameln** [comment]
Hi, I recently answered a similar question in an issue submitted by another user: https://github.com/openvinotoolkit/anomalib/issues/534#issuecomment-1235141974. Does this comment answer your question as well? If not, please let me know which part is still unclear.

**Comment by J-byte-cloud** [comment]
I have another question.why not use train_data to determine early stopping intead of val_data?I think that can reduce the complexity of data set division. @ @djdameln

**Comment by djdameln** [comment]
There are multiple reasons for this. First and most importantly, it is in general bad practice to use the training set for early stopping. Since the model learns directly from the images in the training set, it is likely that the training loss will just keep decreasing as far as the model's capacity allows, which will lead to overfitting.

The second reason is that anomaly models are trained on normal images only, so the training set does not contain any anomalous images. This means that the performance metrics, which are used as the early stopping criterion, cannot be computed over the training set. Similarly, we would not be able to compute the optimal anomaly score threshold when we only have access to normal images. This is why we need a validation set consisting of both normal and abnormal images, in addition to our training set which contains only normal images.

Finally, as explained in https://github.com/openvinotoolkit/anomalib/issues/534#issuecomment-1235141974, a separate test set is needed to compute the final performance of the model after early stopping and computing the adaptive threshold. This test set must consist solely of images that have not been seen by the model during training and validation.

**Comment by J-byte-cloud** [comment]
Thank you so much for patient explanation.I have got it.
